packages/core/src/render3/styling/styling_parser.ts

```diff
@@ -0,0 +1,158 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {CharCode} from '../../util/char_code';
+
+/**
+ * Returns index of next non-whitespace character.
+ *
+ * @param text Text to scan
+ * @param startIndex Starting index of character where the scan should start.
+ * @param endIndex Ending index of character where the scan should end.
+ * @returns Index of next non-whitespace character (May be the same as `start` if no whitespace at
+ *          that location.)
+ */
+export function consumeWhitespace(text: string, startIndex: number, endIndex: number): number {
+  while (startIndex < endIndex && text.charCodeAt(startIndex) <= CharCode.SPACE) {
+    startIndex++;
+  }
+  return startIndex;
+}
+
+/**
+ * Returns index of last char in class token.
+ *
+ * @param text Text to scan
+ * @param startIndex Starting index of character where the scan should start.
+ * @param endIndex Ending index of character where the scan should end.
+ * @returns last char in class token (May be the same as `start` if no non-whitespace
+ *          at that location, or if `startIndex === endIndex`.)
+ */
+export function consumeClassToken(text: string, startIndex: number, endIndex: number): number {
+  while (startIndex < endIndex && text.charCodeAt(startIndex) > CharCode.SPACE) {
+    startIndex++;
+  }
+  return startIndex;
+}
+
+/**
+ * Consumes all of the characters belonging to style key and token.
+ *
+ * @param text Text to scan
+ * @param startIndex Starting index of character where the scan should start.
+ * @param endIndex Ending index of character where the scan should end.
+ * @returns Index of next non-whitespace character (May be the same as `start` if no non-whitespace
```

Comment:
Is this right? From the name, it sounds like this should return the last index of the style key
```suggestion
 * @returns Index after last style key character 
```

Reply:
fixed